blog/2024-11-11-package.md

```diff
@@ -0,0 +1,141 @@
+
+### Must-Follow Guidelines
+
+When creating a new package, ensure that you follow these guidelines:
+
+- **README.md**: Include a `README.md` file in the root directory. This should provide an overview of the package, its purpose, a description of the LF files in the `lib/` folder, and relevant user information. The `README.md` is essential for publishing your package to the community repository and must be kept up to date.
+
+- **File Naming**: Use descriptive names for files in the `lib/` directory that reflect their purpose. For example, name a file containing a reactor for different sensors as `Sensor.lf`.
+
+- **Example LF Files**: Include example LF files in the `src/` directory to demonstrate how to use the reactors in the `lib/` directory. These examples should be well-documented with comments explaining the purpose and usage of each reactor.
+
+## Configuring the `Lingo.toml` File
+
+The `Lingo.toml` file is a configuration file that provides essential information about your package, such as its name, version, and dependencies. It ensures that your package is recognized and installed correctly by the Lingo Package Manager. The `Lingo.toml` file should include the following sections:
+
+- **`[package]`**: Specify the package name, description, and version.
+
+- **`[lib]`**: Define the package name, the target language used to implement the code, the main reactor (which could be one of the examples from the `src/` directory), and the platform used to build the package.
+
+Here's an example of a `Lingo.toml` file:
+
+```toml
+[package]
+name = "PackageName"
+version = "0.3.1"
+
+[properties]
+
+[lib]
+name = "PackageName"
+main = "./src/Main.lf"
+target = "Python"
+platform = "Native"
+
+[lib.properties]
+
+[dependencies]
+```
+
+The `Lingo.toml` can also include additional sections for dependencies and other purpose-specific configurations, however, these are optional and can be added as needed.
+
+## Publishing to the Community Repository
+
+Once you've created your package and configured the `Lingo.toml` file, you’re ready to publish it in the [Lingua Franca Packages](https://github.com/lf-pkgs) organization. Publishing your package here allows other developers to easily find, install, and use it in their projects.
```

Comment:
I think the default workflow should be the folks publish to their own (public) repo. That will already allow for the exchange of reusable libraries. Then, later on in the description we could mention the `lf-pkgs` organization and getting people's repos transfered. I'm actually starting to doubt that this is the right approach at all. I think it might be better to have a `lf-lang/pkgs` repo with a single text file that lists all the repos. That would be super easy to build a little website around where people can search for packages (or a feature in VS Code, for that matter). In the near term, adding a package would just amount to adding the repo URL, and it would not require transferring ownership. Ownership transfer is tedious because anyone who transfers ownership would have to be a member of the `lf-lang` organization, which has paid seats, so this won't scale. Perhaps this blog should just omit these details? Also tagging @tanneberger for feedback on this.

Reply:
> I think it might be better to have a lf-lang/pkgs repo with a single text file that lists all the repos.

I agree with that, it will be easier to show the list of packages, and people in the community won’t need to ask to join the GitHub organization.

So, in this blog, I’ll leave out all the comments about the GitHub organization for now

Reply:
On reflection, if we do that route, we don't even need a separate organization. We could just have a `lf-lang/pkgs` repo.